harness/tests/experiment/pytorch/test_deepspeed_autotuning.py

```diff
@@ -65,55 +56,100 @@ def test_autotuning_module_transforms(
     assert submitted_single_searcher_config_dict.get("searcher", {}).get("name", "") == "single"
 
 
-def test_simple_search_method_happy_path() -> None:
+def _run_searcher(search_method: BaseDSATSearchMethod, all_metrics):
+    """
+    Run a mocked version of the Determined master with a deterministic series of
+    returned metrics for a given Deepspeed Autotune Custom Search Method
+    """
     model_dir = BASE_EXPERIMENT_FIXTURE_PATH.joinpath("example_experiment")
     config_path = model_dir.joinpath("autotune_config.yaml")
     submitted_config_dict = _utils.get_dict_from_yaml_or_json_path(config_path)
 
     with tempfile.TemporaryDirectory() as searcher_dir:
         searcher_dir = pathlib.Path(searcher_dir)
-        search_method = SimpleDSATSearchMethod(
+        search_method = search_method(
             submitted_config_dict=submitted_config_dict,
             model_dir=model_dir,
         )
-        mock_master_obj = MockMaster(
-            all_metrics=[MODEL_INFO_PROFILE_METRIC_FIXTURE, {"throughput": 1.0}]
-        )
+        mock_master_obj = MockMaster(all_metrics=all_metrics)
         search_runner = MockMasterSearchRunner(search_method, mock_master_obj, searcher_dir)
         search_runner.run(exp_config={}, context_dir="", includes=None)
-
-    # TODO: Use a more dynamic value if/when we enable users to configure this
-    exp_num_trials = _defaults.AUTOTUNING_DICT["tuner_num_trials"]
-    assert len(search_runner.state.trials_created) == exp_num_trials
-    assert len(search_runner.state.trials_closed) == exp_num_trials
-    assert len(search_runner.state.trial_progress) == exp_num_trials
-    # TODO: Handle the progress being 6 every time...
-    # for trial_uuid in search_runner.state.trial_progress:
-    #     assert(search_runner.state.trial_progress[trial_uuid] == 1.0)
-
-
-# TODO: Check that the trial tracker stops early when we trigger the early stopping criteria
-
-
-def test_random_search_method() -> None:
-    pass
-
-
-def test_trial_tracker() -> None:
-    pass
-
-
-def test_mbs_binary_search() -> None:
-    pass
+    return search_runner
+
+
+@pytest.mark.timeout(5)
+def test_deepspeed_autotune_happy_path() -> None:
+    """
+    Simulate the Deepspeed Autotune Search Methods end to end and make sure
+    nothing falls over
+    """
+    for search_method in _defaults.ALL_SEARCH_METHOD_CLASSES.values():
+        search_runner = _run_searcher(
+            search_method, [MODEL_INFO_PROFILE_METRIC_FIXTURE, {"throughput": 1.0}]
+        )
+        exp_num_trials = _defaults.AUTOTUNING_DICT["tuner_num_trials"]
+        assert len(search_runner.state.trials_created) == exp_num_trials
+        assert len(search_runner.state.trials_closed) == exp_num_trials
+        assert len(search_runner.state.trial_progress) == exp_num_trials
+        for trial_uuid in search_runner.state.trial_progress:
+            assert search_runner.state.trial_progress[trial_uuid] == 1.0
```

Comment:
I guess you resolved that thing where some Trials had a progress of `6.0`? Or was that elsewhere?

Reply:
I changed that in the MockMaster actually. The training length isn't valid in this particular case as far as I understand